pkg/cloud/awscloud/uploader.go

```diff
@@ -22,10 +24,30 @@ type awsUploader struct {
 	bucketName string
 	imageName  string
 	targetArch string
+	bootMode   *string
```

Comment:
Is seems that bootMode in platform is never an empty string so I wonder if we could simply use `string` here and `""` instead of a pointer (not a fan of pointers). But I guess it would require quite a few more change(?). I could look into it for a followup maybe? (if desired)

Reply:
You're right, it should be fine.
Also, the 0 value for the internal enum `BOOT_NONE` is `"none"`, though that's probably not entirely relevant here.

Reply:
Yeah, I'm considering either using `BOOT_NONE` or introducing `BOOT_UNSET` as the `0` value. I still need to check the potential impact.